tests/lib/tools/tests.backup

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: tests.backup prepare"
+    echo "       tests.backup restore"
+}
+
+cmd_prepare() {
+    local BACKUP_PATH=${1:-$(pwd)}
+
+    if [ ! -d "$BACKUP_PATH" ]; then
+        echo "tests.backup: cannot prepare, backup dir does not exist" >&2
+        exit 1
+    fi
+    tar cf "${BACKUP_PATH}.tar" "$BACKUP_PATH"
+}
+
+cmd_restore() {
+    local BACKUP_PATH=${1:-$(pwd)}
+    if [ -f "${BACKUP_PATH}.tar" ]; then
+        find "${BACKUP_PATH}" -maxdepth 1 -mindepth 1 -exec rm -rf {} \;
```

Comment:
How does this differ from `rm -rf "$BACKUP_PATH"; mkdir "$BACKUP_PATH`"?

Reply:
Please see the comment from @mvo https://github.com/snapcore/snapd/pull/9425#discussion_r496567494

Reply:
I'm sorry can you re-phrase that in a code comment. I think I'm missing the point and this is an important property that ought to be documented in the code.